### ISSUE TITLE
Comparison between version yolov8 and yolov7

### ISSUE THREAD

**Issue by sarpx** [ultralytics/ultralytics#1182]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I trained the same two data with versions 8 and 7 with very different results. I wasn't expecting that much difference. Is there a setting missing?

v7
---
python segment/train.py --batch 14 --epochs 300 --data ./girdi/v5001_1675781502310.yaml --cfg ./models/segment/yolov5x-seg.yaml --weights ./weights/yolov5x-seg.pt --img 640 --device 0 --exist-ok --name ./faz030123/v5001_1675781502310

v8
----
yolo segment train model=./ek/yolov8x-seg.pt data=../yolov5-7.0/girdi/v5001_1675781502310.yaml device=0 batch=14 epochs=300  imgsz=640 name=./faz030123/v5001_8_1675781502310

### Additional
results v7
[v7.txt](https://github.com/ultralytics/yolov5/files/10719314/v7.txt)
-----
results v8
[v8.txt](https://github.com/ultralytics/yolov5/files/10719315/v8.txt)
-----
![results_V8](https://user-images.githubusercontent.com/73984193/218387176-5c62cbe9-22cc-4bd1-809b-9d610cabd511.png)
![results_v7](https://user-images.githubusercontent.com/73984193/218387300-18a689c9-d5aa-47f3-932e-ff6df9efc5f1.png)

**Comment by 35grain** [comment]
It looks like you are using v5 weights instead of v7 in the v7 command:
`...--cfg ./models/segment/yolov5x-seg.yaml --weights ./weights/yolov5x-seg.pt...`

**Comment by pderrenger** [comment]
@sarpx it seems that you are encountering different results between YOLOv8 and YOLOv7 when training with the same data, and it appears that you are using the correct parameters for YOLOv8.

Given this situation, have you considered performing a detailed comparison between the architecture and hyperparameters of YOLOv7 and YOLOv8 to identify potential factors contributing to the observed differences in performance? This may involve examining changes in the model architecture, training parameters, and other aspects that may have been updated between these two versions. Additionally, conducting an in-depth analysis of the training process in both versions could provide valuable insights into the discrepancies in performance.

Feel free to share any further details or specific aspects that you would like to focus on, and I'd be happy to assist you in addressing this issue effectively.

**Comment by colourpalettes3423** [comment]
Interesting comparison! YOLOv7 is still appreciated for its impressive speed and accuracy, especially in real-time tasks with limited hardware. However, this [YOLOv8 comparison](https://yolov8.org/yolov8-comparison/) highlights some key upgrades like a cleaner architecture, built-in multi-task support, and improved training stability. One issue often discussed is that while YOLOv7 can outperform v8 in pure FPS, YOLOv8 offers more flexibility and better long-term maintainability. Choosing between them really depends on the specific deployment scenario.

**Comment by glenn-jocher** [comment]
@sarpx the performance differences you're seeing between YOLOv7 and YOLOv8 are expected due to significant architectural changes between these models. YOLOv8 introduced several key improvements including an [anchor-free detection head](https://docs.ultralytics.com/compare/yolov7-vs-yolov8/) versus YOLOv7's anchor-based approach, and replaced the C3 module with the more efficient C2f module for better feature fusion.

For a detailed technical analysis of these architectural differences and their impact on performance, I'd recommend checking our [YOLOv7 vs YOLOv8 comparison guide](https://docs.ultralytics.com/compare/yolov7-vs-yolov8/). YOLOv8 typically delivers higher accuracy (mAP) while maintaining competitive speeds, but the specific performance characteristics can vary depending on your dataset and use case.
